CartingApp/Cart.cs

```diff
@@ -6,49 +6,71 @@ namespace CartingApp
 {
     public class Cart
     {
-        public List<CartItem> cartItemList;
-        public Bill bill;
+        public List<CartItem> cartItemList { get; private set; }
+        public Invoice invoice;
 
-        public Cart(DiscountType discountType)
+        public Cart(DiscountType discountType, double discount)
         {
-            bill = new Bill(discountType);
             cartItemList = new List<CartItem>();
+            invoice = new Invoice(this.cartItemList, DiscountType.CategoryDiscount);
         }
 
-        public void AddProduct(CartItem cartItem)
+        public void AddCartItem(CartItem cartItem)
         {
             cartItemList.Add(cartItem);
         }
 
-        public void RemoveProduct(string productName)
+        public void RemoveCartItem(string productName)
         {
-            cartItemList.Remove(
-                cartItemList.Find(x => x.product.name == productName)
+            try
+            {
+                cartItemList.Remove(
+                    cartItemList.Find(x => x.product.name == productName)
                 );
+            }
+            catch (System.Exception)
+            {
+
+            }
+        }
+
+        public CartItem GetCartItem(string productName)
+        {
+            try
+            {
+                return cartItemList.Find(x => x.product.name == productName);
+
+            }
+            catch (System.Exception)
+            {
+                return null;
+            }
         }
 
         public List<string> ListCartItemProductName()
         {
             List<string> cartProductNames = new List<string>();
+
             cartItemList
                 .ForEach(
                     x => cartProductNames.Add(x.product.name)
                 );
-
             return cartProductNames;
         }
 
         public void ChangeProductQuantity(string productName, int quantity)
         {
-            cartItemList
+            try
```

Comment:
Add appropriate error handling for negative quantity and check if product already exists or not